src/shadowbox/server/manager_service.spec.ts

```diff
@@ -278,11 +276,15 @@ describe('ShadowsocksManagerService', () => {
       };
       const next = (error) => {
         expect(error.statusCode).toEqual(400);
-        responseProcessed = true;
-        done();
       };
 
       await service.setPortForNewAccessKeys(noPort, res, next);
+      await service.setPortForNewAccessKeys(
```

Comment:
Can you add a test case where port is passed as a string in the request?

Reply:
There is one, the test case on line 262 'rejects malformed requests'.  See `nonNumericPort`.